src/twilio/functions/sms/test-delivery.js

```diff
@@ -0,0 +1,102 @@
+// this endpoint version receives delivery/intake name + phone numbers from the React web app
```

Comment:
This seems to be a duplicate of intake-delivery-txts, i think you can use the same code as intake-delivery-txts, but replace any staging values with a context variable, like how `context.AIRTABLE_API_KEY` is passed in. I don't know much about how env vars are passed in Twilio, but happy to figure it out together if you'd like!

Having less duplicate code will help reduce work and mistakes in the future when we need to update this code!

Reply:
didn't get to this yet, but i think i know what you mean: we can have 1 twilio function, so just the 1 endpoint URL that would work for both staging and prod, that switches only which Airtable base we read. 

what's my `if` statement, then - how do I determine if it's staging? is it the heroku URL, do I ask where the request is coming from?

thnx!

Reply:
ah i meant that it can be same code, but still two different twilio endpoints. one endpoint can be the staging and the other prod, but you deploy the same code, but set different `context.AIRTABLE_BASE`. is that possible for twilio? 

if not, we can keep two different files, but maybe abstract out the main logic. so prod and staging  call the same helper function. e.g.:

sendDeliveryTexts.js
```
function sendDeliveryTexts(airtableBase){
  // set given airtable base here
  ...
}

```

intake-delivery-texts-prod.js
```
sendDeliveryTexts(<prod airtable base>)
```

intake-delivery-texts-staging.js
```
sendDeliveryTexts(<staging airtable base>)
```

either way is fine, as long as we can minimize duplicate code!

also, just a nit, but wondering if we can use the same name for the endpoint, just with prod/staging? e.g. /intake-delivery-texts-prod and /intake-delivery-texts-staging. it will be easier for us to keep track of which is which this way!

Reply:
ah i see!

yeah so how twilio creates the endpoints is with each file.  i realize they're totally duplicate but since they're stateless functions, it only reads from what's in that 1 file ... i think the only way to avoid totally duplicating would actually to have 3 files:

intake-delivery-texts-prod -> sets the base variable for prod, then calls /send-intake-delivery-texts endpoint
intake-delivery-texts-staging -> sets the base variable for staging, then calls /send-intake-delivery-texts endpoint
send-intake-delivery-texts -> knows the base from the firing event, but has the meat of the code

does that make sense? if it's ok to duplicate the code to have 2 endpoints only, i think that might be easier ..!

Reply:
Oh I see, is it not possible to import a helper function with `require`? 

Since `require('airtable') works`, i wonder if you can just do 

```
const sendDeliveryTexts = require('./sendDeliveryTexts')
```

 where `sendDeliveryTexts.js` is the helper function file with main logic. 

Or did you have to do something special to enable require for Airtable? I'm not super familiar with stateless functions, let me know if that's not possible! Just throwing suggestions out there.

Reply:
hmmmm i would have to test this! it's a good idea, i've just never tried ... 
[i know that the require("airtable") is an npm package, but that doesn't necessarily mean i couldn't require other scripts that are hosted in the twilio environment!]

not sure i could get to this today ... but probably by monday/tuesday?!

Reply:
yeah no worries! i think we are ok to merge without that change, can you make a GH issue for this refactor so it gets tracked? looking forward to having this in the repo!!

Reply:
yassssss post-merge refactor makes sense, we can dig into how to make it elegant